modules/puppetdb_fact/plans/init.pp

```diff
@@ -0,0 +1,7 @@
+plan puppetdb_fact(TargetSpec $nodes) {
+  $certnames = get_targets($nodes).map |$target| { $target.uri }
```

Comment:
I think you want `$target.host`. `uri` will frequently not be a valid certname because it can include protocol/port/etc.